tensorflow/python/ops/random_ops.py

```diff
@@ -313,18 +314,30 @@ def random_crop(value, size, seed=None, name=None):
     value = ops.convert_to_tensor(value, name="value")
     size = ops.convert_to_tensor(size, dtype=dtypes.int32, name="size")
     shape = array_ops.shape(value)
-    check = control_flow_ops.Assert(
-        math_ops.reduce_all(shape >= size),
-        ["Need value.shape >= size, got ", shape, size],
-        summarize=1000)
-    shape = control_flow_ops.with_dependencies([check], shape)
-    limit = shape - size + 1
-    offset = random_uniform(
-        array_ops.shape(shape),
-        dtype=size.dtype,
-        maxval=size.dtype.max,
-        seed=seed) % limit
-    return array_ops.slice(value, offset, size, name=name)
+
+    with tf.Session():
+        size_dim = size.eval()
```

Comment:
This has a cost to evaluate, I'm not sure its a good idea to do it here

Reply:
I see. I'll find another way.